src/scripts/main.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
+const form = document.querySelector('form');
+
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (element) {
+      element.addEventListener(eventName, () => {
+        const string = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
```

Comment:
The string you're creating here is a bit hard to read. Consider using template literals to make it more readable.